include/mxnet/lib_api.h

```diff
@@ -18,33 +18,756 @@
  */
 
 /*!
- * Copyright (c) 2015 by Contributors
+ * Copyright (c) 2019 by Contributors
  * \file lib_api.h
  * \brief APIs to interact with libraries
+ * This API specifies function prototypes to
+ * register custom ops for library authors
  */
+
 #ifndef MXNET_LIB_API_H_
 #define MXNET_LIB_API_H_
 
+#include <stdint.h>
+#include <stdlib.h>
+#include <vector>
+#include <map>
+#include <string>
+#include <iostream>
+#include <utility>
+
+#define MX_LIBRARY_VERSION 1
+
+/*!
+ * \brief External Tensor data types
+ */
+enum MXDType {
```

Comment:
Is it consistent with DLPack or mshadow?

Reply:
this is consistent with mshadow, not dlpack